provider/cmd/app/app.go

```diff
@@ -453,3 +477,42 @@ func runStreamingServer(ctx context.Context, setupLog, log logr.Logger, srv *ser
 	}
 	return nil
 }
+
+func runMetricsServer(ctx context.Context, setupLog logr.Logger, opts HTTPServerOptions) error {
+	if opts.Addr == "" {
+		setupLog.Info("Metrics server address isn't configured. Metrics server is disabled.")
+		return nil
+	}
+
+	setupLog.Info("Starting metrics server on " + opts.Addr)
+
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", promhttp.Handler())
+
+	srv := http.Server{
+		Addr:         opts.Addr,
+		Handler:      mux,
+		ReadTimeout:  opts.ReadTimeout,
+		WriteTimeout: opts.WriteTimeout,
+		IdleTimeout:  opts.IdleTimeout,
+	}
+
+	go func() {
+		<-ctx.Done()
+		setupLog.Info("Shutting down streaming server")
+		shutdownCtx, cancel := context.WithTimeout(context.Background(), time.Second)
+		defer cancel()
+		locErr := srv.Shutdown(shutdownCtx)
+		if locErr != nil {
+			setupLog.Error(locErr, "metrics server wasn't shutdown properly")
+		}
+		setupLog.Info("Metrics server is shutdown")
```

Comment:
Here, even if the Shutdown function returns an error, the statement following the error handling will still be executed. This could be misleading as it implies that the server shutdown was successful, even if an error occurred.

Reply:
i will move it to line 516, it will be better

Reply:
fixed. I did shutdown and logging more flexible.